sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -115,6 +115,108 @@ def process(self, element):
       for _ in range(self._output_records_per_input_record):
         yield element
 
+class SyntheticSDFStepRestrictionProvider(RestrictionProvider):
+  """A `RestrictionProvider` for SyntheticSDFStep.
+
+  In initial_restriction and split that operates on num_records and ignores
+  source description (element).
+
+  """
+
+  def __init__(self, num_records, initial_splitting_num_bundles):
+    self._num_records = num_records
+    self._initial_splitting_num_bundles = initial_splitting_num_bundles
+
+  def initial_restriction(self, element):
+    return (0, self._num_records)
```

Comment:
Can we return an OffsetRange instead of a tuple?

Reply:
Any reason why prefer using OffRange rather than a (start_pos, end_pos)? If using OffsetRange instead of a simple tuple, then a RestrictionProvider can only create an OffRangeTracker in process fn, right? I'm not sure whether there are other links of tracker expected to be created, though.

Reply:
The SDF is not required to use an OffsetRangeTracker. Any restriction tracker that uses OffsetRange as the restriction type is valid.

Reply:
Sounds good.